image_segmentation/pytorch/runtime/inference.py

```diff
@@ -0,0 +1,153 @@
+import numpy as np
+from scipy import signal
+from tqdm import tqdm
+
+import torch
+import torch.nn.functional as F
+from torch.cuda.amp import autocast
+
+from runtime.distributed_utils import reduce_tensor, get_world_size, get_rank
+
+
+def evaluate(flags, model, loader, loss_fn, score_fn, device, epoch=0, is_distributed=False):
+    rank = get_rank()
+    world_size = get_world_size()
+    model.to(device)
+    if flags.load_ckpt_path:
+        map_location = {'cuda:%d' % 0: 'cuda:%d' % rank}
+        checkpoint = torch.load(flags.load_ckpt_path, map_location=map_location)
+        epoch = checkpoint['epoch']
+        model.load_state_dict(checkpoint['best_model_state_dict'])
+        if is_distributed:
+            model = torch.nn.parallel.DistributedDataParallel(model,
+                                                              device_ids=[flags.local_rank],
+                                                              output_device=flags.local_rank)
+
+    model.eval()
+
+    eval_loss = []
+    scores = []
+    with torch.no_grad():
+        for i, batch in enumerate(tqdm(loader, disable=(rank != 0) or not flags.verbose)):
+            image, label = batch
+            image, label = image.to(device), label.to(device)
+            if image.numel() == 0:
+                continue
+            with autocast(enabled=flags.amp):
+                output, label = sliding_window_inference(
+                    inputs=image,
+                    labels=label,
+                    roi_shape=flags.input_shape,
```

Comment:
Should this be val_input_shape or input_shape?

Reply:
They are the same but yeah, good point.